[bug]: failed to send payments in `zero_conf-channel_policy_update_public_zero_conf`

Found from this [build](https://github.com/lightningnetwork/lnd/actions/runs/15445236160/job/43472922533?pr=9880),
```
--- FAIL: TestLightningNetworkDaemon/tranche01/48-of-301/bitcoind/zero_conf-channel_policy_update_public_zero_conf (151.80s)
        harness_node.go:403: Starting node (name=Eve) with PID=18006
        harness_node.go:403: Starting node (name=Carol) with PID=18306
        harness_node.go:403: Starting node (name=Dave) with PID=18496
        harness_assertion.go:1087: 
            	Error Trace:	/home/runner/work/lnd/lnd/lntest/harness_assertion.go:1087
            	            				/home/runner/work/lnd/lnd/lntest/harness_assertion.go:1040
            	            				/home/runner/work/lnd/lnd/lntest/harness.go:1588
            	            				/home/runner/work/lnd/lnd/lntest/harness.go:1601
            	            				/home/runner/work/lnd/lnd/itest/lnd_zero_conf_test.go:673
            	            				/home/runner/work/lnd/lnd/itest/lnd_zero_conf_test.go:65
            	            				/home/runner/work/lnd/lnd/lntest/harness.go:303
            	            				/home/runner/work/lnd/lnd/itest/lnd_test.go:130
            	Error:      	Received unexpected error:
            	            	received error from payment stream: EOF
            	Test:       	TestLightningNetworkDaemon/tranche01/48-of-301/bitcoind/zero_conf-channel_policy_update_public_zero_conf
            	Messages:   	timeout while waiting payment
        harness.go:381: finished test: zero_conf-channel_policy_update_public_zero_conf, start height=574, end height=588, mined blocks=14
        harness.go:340: test failed, skipped cleanup
    lnd_test.go:138: Failure time: 2025-06-04 15:32:32.374
```

Eve tried to send a payment to Dave via Carol, but Carol cannot find the channel between her and Dave, causing the payment to fail,
```
2025-06-04 15:31:01.819 [DBG] HSWC link.go:4115: ChannelLink([ChanPoint: Eve=>Carol]): forwarding 1 packets to switch: replay=false
2025-06-04 15:31:01.822 [DBG] HSWC switch.go:2253: Querying outgoing link using chanID=583:1:0, aliasID=false
2025-06-04 15:31:01.822 [DBG] HSWC switch.go:2869: unable to find link with destination 583:1:0
2025-06-04 15:31:01.822 [ERR] HSWC switch.go:1238: UnknownNextPeer
2025-06-04 15:31:01.822 [ERR] HSWC switch.go:840: Unhandled error while reforwarding htlc settle/fail over htlcswitch: UnknownNextPeer
2025-06-04 15:31:01.822 [DBG] HSWC link.go:1926: ChannelLink([ChanPoint: Eve=>Carol]): queueing removal of FAIL closed circuit: (Chan ID=577:1:0, HTLC ID=0)->(Chan ID=583:1:0, HTLC ID=0)
```

It's like a bug in zeroconf or channel policy update, causing the channel to not be found here,
https://github.com/lightningnetwork/lnd/blob/4a7cd0095e37c2caf080d39d2f7290a5d38d831e/htlcswitch/switch.go#L2285-L2306

--- a/funding/manager.go
+++ b/funding/manager.go
@@ -3957,6 +3957,22 @@ func (f *Manager) waitForZeroConfChannel(c *channeldb.OpenChannel) error {
 
 	// Six confirmations have been reached. If this channel is public,
 	// we'll delete some of the alias mappings the gossiper uses.
+	//
+	// Tell the Switch to refresh the relevant ChannelLink so that forwards
+	// under the confirmed SCID are possible. We do this BEFORE updating the
+	// graph to avoid a race where other nodes learn about the confirmed
+	// SCID from gossip before our switch is ready to handle forwards using
+	// it. This is especially important for integration tests.
+	err = f.cfg.ReportShortChanID(c.FundingOutpoint)
+	if err != nil {
+		// This should only fail if the link is not found in the
+		// Switch's linkIndex map. If this is the case, then the peer
+		// has gone offline and the next time the link is loaded, it
+		// will have a refreshed state. Just log an error here.
+		log.Errorf("unable to report scid for zero-conf channel "+
+			"channel: %v", err)
+	}
+
 	isPublic := c.ChannelFlags&lnwire.FFAnnounceChannel != 0
 	if isPublic {
 		err = f.cfg.AliasManager.DeleteSixConfs(c.ShortChannelID)
@@ -3983,19 +3999,6 @@ func (f *Manager) waitForZeroConfChannel(c *channeldb.OpenChannel) error {
 			return fmt.Errorf("failed adding confirmed zero-conf "+
 				"SCID to graph: %v", err)
 		}
-	}
-
-	// Since we have now marked down the confirmed SCID, we'll also need to
-	// tell the Switch to refresh the relevant ChannelLink so that forwards
-	// under the confirmed SCID are possible if this is a public channel.
-	err = f.cfg.ReportShortChanID(c.FundingOutpoint)
-	if err != nil {
-		// This should only fail if the link is not found in the
-		// Switch's linkIndex map. If this is the case, then the peer
-		// has gone offline and the next time the link is loaded, it
-		// will have a refreshed state. Just log an error here.
-		log.Errorf("unable to report scid for zero-conf channel "+
-			"channel: %v", err)
 	}
 
 	// Update the confirmed transaction's label.

--- a/funding/manager_test.go
+++ b/funding/manager_test.go
@@ -4726,14 +4726,9 @@ func testZeroConf(t *testing.T, chanType *lnwire.ChannelType) {
 	assertConfirmationHeight(t, alice, chanID, 1)
 	assertConfirmationHeight(t, bob, chanID, 1)
 
-	// For taproot channels, we don't expect them to be announced atm.
-	if !isTaprootChanType(chanType) {
-		assertChannelAnnouncements(
-			t, alice, bob, fundingAmt, nil, nil, nil, nil,
-		)
-	}
-
-	// Both Alice and Bob should send on reportScidChan.
+	// Both Alice and Bob should call ReportShortChanID first (before
+	// sending announcements) to avoid a race where other nodes learn about
+	// the confirmed SCID before the switch is ready.
 	select {
 	case <-alice.reportScidChan:
 	case <-time.After(time.Second * 5):
@@ -4744,6 +4739,13 @@ func testZeroConf(t *testing.T, chanType *lnwire.ChannelType) {
 	case <-bob.reportScidChan:
 	case <-time.After(time.Second * 5):
 		t.Fatalf("did not call ReportShortChanID in time")
+	}
+
+	// For taproot channels, we don't expect them to be announced atm.
+	if !isTaprootChanType(chanType) {
+		assertChannelAnnouncements(
+			t, alice, bob, fundingAmt, nil, nil, nil, nil,
+		)
 	}
 
 	// Send along the 6-confirmation channel so that announcement sigs can

